azure/test.yml

```diff
@@ -30,7 +32,23 @@ jobs:
     parameters:
       rust: ${{ parameters.rust }}
       setup: ${{ parameters.setup }}
-  - script: cargo test --all
-    displayName: Run tests
-    env:
-      ${{ insert }}: ${{ parameters.envs }}
+  - ${{ if and(eq(parameters.single_threaded, 'true'), eq(parameters.test_ignored, 'true')) }}:
+    - script: cargo test --all -- --ignored --test-threads=1
```

Comment:
Could we use the conditions to build up a parameter list somehow, like in an environment variable, to simplify the logic?

Reply:
I wish, but sadly I don't see a good way of doing that given Azure's particular templating restrictions. If you can think of a way, I'm all ears!

Reply:
Bah, I hate digging into the Pipelines documentation.  I know there are ways for creating environment variables but can't find the docs to see if its flexible enough for our needs.

Reply:
We might be able to play som trickery with `bash:` and whitespace splitting to construct the argument list, but then I _think_ this is cleaner if possibly more verbose.

Reply:
Actually, this is another reason it'd be nice to have a script that _generates_ the `.yml` files. Then this'd be trivial to do.